pp/series_data/unloading/unloader.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include "bare_bones/bit.h"
+#include "bare_bones/bitset.h"
+#include "bare_bones/encoding.h"
+#include "series_data/data_storage.h"
+#include "series_data/encoder/bit_sequence.h"
+
+namespace series_data::unloading {
+class Unloader {
+  using EncodingChunkLengthSequence =
+      BareBones::EncodedSequence<BareBones::Encoding::DeltaDeltaZigZag<BareBones::StreamVByte::CompactSequence<BareBones::StreamVByte::Codec0124Frequent0>>>;
+  using EncodingChunkIDSequence =
+      BareBones::EncodedSequence<BareBones::Encoding::RLE<BareBones::StreamVByte::CompactSequence<BareBones::StreamVByte::Codec0124Frequent0>>>;
+
+ public:
+  explicit Unloader(DataStorage& storage) : storage_(storage) {}
+
+  template <class Stream>
+  void unload(Stream& stream) {
+    const auto sequences = prepare_sequences();
+    write_sequences(stream, sequences.ls_id_bitmap, sequences.chunk_length_sequence, sequences.chunk_id_sequence);
+    write_bit_sequences(stream, sequences.ls_id_bitmap, sequences.total_bitseqs_size);
+  }
+
+  static constexpr uint32_t get_empty_unloader_size_in_bytes() noexcept {
+    return sizeof(uint32_t) + BareBones::Bitset{}.allocated_memory() + EncodingChunkLengthSequence{}.data().size() + sizeof(uint32_t) + sizeof(uint32_t) +
+           EncodingChunkIDSequence{}.data().size() + sizeof(uint32_t) + sizeof(uint32_t) + sizeof(uint32_t);
+  }
+
+ private:
+  DataStorage& storage_;
+
+  struct PreparedSequences {
+    BareBones::Bitset ls_id_bitmap;
+    EncodingChunkLengthSequence chunk_length_sequence;
+    EncodingChunkIDSequence chunk_id_sequence;
+    uint32_t total_bitseqs_size;
+  };
+
+  [[nodiscard]] PreparedSequences prepare_sequences() const noexcept {
+    PreparedSequences result{};
+    result.total_bitseqs_size = 0;
+
+    for (const auto ls_id : storage_.unused_series_bitmap) {
+      const auto encoding_type = storage_.open_chunks[ls_id].encoding_state.encoding_type;
+      if (!storage_.open_chunks[ls_id].is_empty() && is_unloadable_encoder(encoding_type)) {
```

Comment:
```suggestion
      if (!storage_.open_chunks[ls_id].is_empty() && is_unloadable_encoder(encoding_type)) {
```
```suggestion
      if (is_unloadable_encoder(encoding_type)) {
```
Because unloadable chunk can't be empty